ledger/simulation/trace.go

```diff
@@ -163,29 +177,41 @@ func makeSimulationResult(lastRound basics.Round, request Request, developerAPI
 	}, nil
 }
 
-// OpcodeTraceUnit contains the trace effects of a single opcode evaluation
+// OpcodeTraceUnit contains the trace effects of a single opcode evaluation.
+// NOTE: we are introducing model.TealValue here, to avoid declare again conversion subroutines in daemon/handlers.go
```

Comment:
Convertion code in the handlers file is annoying, but I do like the separation of concerns it grants us.

By that I mean the simulation package doesn't need to concern itself with how its structs will be serialized/deserialized and match up with our OAS2 API spec.

Reply:
Yikes.  I was indeed being lazy on previous implementation, and my bad for mashing the API stuffs and simulation logic together.

See [use conversion subroutine in daemon, separation from simulate package](https://github.com/algorand/go-algorand/pull/5440/commits/61d7313976ab484a9e07cab7ee99bccc14dbf151) for the separation.